src/Shared/FileUtilities.cs

```diff
@@ -223,6 +223,20 @@ internal static string EnsureNoTrailingSlash(string path)
             return path;
         }
 
+        /// <summary>
+        /// Ensures the path does not have a leading or trailing slash.
+        /// </summary>
+        internal static string EnsureNoLeadingOrTrailingSlash(string path)
+        {
+            if (String.IsNullOrEmpty(path))
+            {
+                return path;
+            }
+            int start = IsSlash(path[0]) ? 1 : 0;
+            int end = IsSlash(path[path.Length - 1]) ? 1 : 0;
+            return path.Length - start - end > 0 ? path.Substring(start, path.Length - start - end) : String.Empty;
```

Comment:
Wondering if there's a more elegant way to express this, the best I could come up with is:

```suggestion
            int start = IsSlash(path[0]) ? 1 : 0;
            if (path.Length == 1)
            {
                return path.Substring(start);
            }
            int end = IsSlash(path[path.Length - 1]) ? 1 : 0;
            return path.Substring(start, path.Length - start - end);
```

Probably not any better, though.